skyportal/handlers/api/tns.py

```diff
@@ -40,90 +48,199 @@
 log = make_log('api/tns')
 
 
-class TNSRobotHandler(BaseHandler):
-    @auth_or_token
-    def get(self, tnsrobot_id=None):
-        """
-        ---
-        single:
-          tags:
-            - tnsrobots
-          description: Retrieve a TNS robot
-          parameters:
-            - in: path
-              name: tnsrobot_id
-              required: true
-              schema:
-                type: integer
-            - in: query
-              name: groupID
-              schema:
-                type: integer
-              description: |
-                Filter by group ID
-          responses:
-            200:
-               content:
-                application/json:
-                  schema: SingleTNSRobot
-            400:
-              content:
-                application/json:
-                  schema: Error
-        multiple:
-          tags:
-            - tnsrobots
-          description: Retrieve all TNS robots
-          responses:
-            200:
-              content:
-                application/json:
-                  schema: ArrayOfTNSRobots
-            400:
-              content:
-                application/json:
-                  schema: Error
-        """
-
-        group_id = self.get_query_argument("groupID", None)
-
-        with self.Session() as session:
-            try:
-                # get owned tnsrobots
-                stmt = TNSRobot.select(session.user_or_token).options(
-                    joinedload(TNSRobot.auto_report_streams),
-                    joinedload(TNSRobot.auto_report_instruments),
+def create_tns_robot(
```

Comment:
Type annotate or add docstrings, even for internal functions. It's surprising how hard it can be to guess later on what we, ourselves, expected as input at the time.

Reply:
Agreed, will add now!